### PR TITLE
add support for setting SecurityContext (user, group) for restic restore

### PR DESCRIPTION
**Pull request by jaygridley**
Partially fixes https://github.com/vmware-tanzu/velero/issues/2290. Adds support only for runAsUser and runAsGroup which is enough for our use-case, but may not be for others.

### CODE REVIEW

**Comment by ashish-amarnath** [comment]
@jaygridley Thank you for this PR and thank you for patiently waiting for us to look at it.
I am looking at this right now.
In the meantime, can you please resolve the merge conflict?

File: site/docs/master/restic.md [hunk 1]
```diff
@@ -298,7 +298,11 @@ data:
   # If not set, it will default to "128Mi". A value of "0" is treated as unbounded.
   memLimit: 128Mi
 
+  # "secCtxRunAsUser sets the securityContext.runAsUser value on the restic init containers during restore."
```
Comment (ashish-amarnath) [inline site/docs/master/restic.md:301]: This file has been moved to `site/docs/main/restic.md`. This is what is flagged as a merge conflict.
Can you please resolve this merge conflict?

**Comment by jaygridley** [comment]
@ashish-amarnath Should be resolved

File: pkg/util/kube/security_context.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2019 the Velero contributors.
```
Comment (carlisia) [inline pkg/util/kube/security_context.go:2]:
```suggestion
Copyright 2020 the Velero contributors.
```

File: pkg/util/kube/security_context_test.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2019 the Velero contributors.
```
Comment (carlisia) [inline pkg/util/kube/security_context_test.go:2]:
```suggestion
Copyright 2020 the Velero contributors.
```

**Comment by carlisia** [comment]
Also, thanks for adding the test!

**Comment by ashish-amarnath** [comment]
@carlisia This PR has been sitting on our queue for a while now. Lets get this merged, I'll do the copyright boilerplate cleanup in a follow-up PR.

**Comment by ashish-amarnath** [comment]
Here is the follow-up PR with the fixes https://github.com/vmware-tanzu/velero/pull/2751

**Pull request merged**